tests/monitoring_test.go

```diff
@@ -0,0 +1,472 @@
+						if rule.Alert != "" {
+							Expect(rule.Labels).ToNot(BeNil())
+							checkForSeverityLabel(rule)
+							checkForHealthImpactLabel(rule)
+							checkForPartOfLabel(rule)
+							checkForComponentLabel(rule)
+						}
+					}
+				}
+			}
+		})
+	})
+})
+
+func dataVolumeUnusualRestartTest(f *framework.Framework) {
+	By("Test metric for unusual restart count")
+	Eventually(func() bool {
+		return getMetricValue(f, "cdi_datavolume_unusual_restart_count") == 1
+	}, 2*time.Minute, 1*time.Second).Should(BeTrue())
+
+	By("checking that the CDIDataVolumeUnusualRestartCount alert is firing")
+	waitForPrometheusAlert(f, "CDIDataVolumeUnusualRestartCount")
+}
+
+// Helper functions
+
+func getMetricValue(f *framework.Framework, endpoint string) int {
+	var returnVal string
+
+	Eventually(func() bool {
+		var result map[string]interface{}
+		resp := f.MakePrometheusHTTPRequest("query?query=" + endpoint)
+		defer resp.Body.Close()
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return false
+		}
+		err = json.Unmarshal(bodyBytes, &result)
+		if err != nil {
+			return false
+		}
+		if len(result["data"].(map[string]interface{})["result"].([]interface{})) == 0 {
+			return false
+		}
+		values := result["data"].(map[string]interface{})["result"].([]interface{})[0].(map[string]interface{})["value"].([]interface{})
+		for _, v := range values {
+			if s, ok := v.(string); ok {
+				returnVal = s
+				return true
+			}
+		}
+		return false
+	}, 1*time.Minute, 1*time.Second).Should(BeTrue())
+
+	i, err := strconv.Atoi(returnVal)
+	Expect(err).ToNot(HaveOccurred())
+	return i
+}
+
+func createUnknownStorageClass(name, provisioner string) *storagev1.StorageClass {
+	immediateBinding := storagev1.VolumeBindingImmediate
+
+	return &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"cdi.kubevirt.io/testing": "",
+			},
+		},
+		Provisioner:       provisioner,
+		VolumeBindingMode: &immediateBinding,
+	}
+}
+
+func getPrometheusRule(f *framework.Framework) *promv1.PrometheusRule {
+	By("Wait for prometheus-cdi-rules")
+	promRule := &promv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "prometheus-cdi-rules",
+			Namespace: f.CdiInstallNs,
+		},
+	}
+	Eventually(func() error {
+		return f.CrClient.Get(context.TODO(), crclient.ObjectKeyFromObject(promRule), promRule)
+	}, 5*time.Minute, 1*time.Second).Should(BeNil())
+	return promRule
+}
+
+func setPrometheusRule(f *framework.Framework, promRule *promv1.PrometheusRule) {
+	Eventually(func() error {
+		err := f.CrClient.Delete(context.TODO(), promRule)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+
+		promRule.ResourceVersion = ""
+		promRule.UID = ""
+		return f.CrClient.Create(context.TODO(), promRule)
+	}, 5*time.Minute, 1*time.Second).Should(BeNil())
+}
+
+func setPrometheusRuleAlertDuration(f *framework.Framework, promRule *promv1.PrometheusRule, alertName, duration string) {
```

Comment:
This is not used along with `setPrometheusRule`, IIRC because the operator would just revert the changes